zod/src/zod.ts

```diff
@@ -69,7 +69,7 @@ export const zodResolver: Resolver =
 
         return {
           errors: {} as FieldErrors,
-          values: data,
+          values: resolverOptions.rawValues ? values : data,
```

Comment:
Unsure if it was ever intentional to pass the parsed output here.

Reply:
Thanks for the PR, it's intentional to return what's provided by the resolvers. What's the issue that brings?

Reply:
I made a reproduction


1. Go to https://kitchen-sink-git-repro-trpc.vercel.app/react-hook-form
2. Open network tab
3. Fill in form

- Expected: submits successfully
- Actual doesn't work


Highlighted inline here https://github.com/trpc/examples-kitchen-sink/pull/36 on why it doesn't work - it's because the zod resolver coerces the values before sending them to the server which will then try to validate them based on the input:

1. `dateStr` gets validated as `Date` on client
2. `Date` is sent to server
3. Server receives `Date`, tries to validate as a string --> ❌

Reply:
thanks for the context. `rawValues` option I couldn't find in zod's doc do you have a reference?

Reply:
With raw just means that we're not using zod's parsed output

Reply:
yea, is that a public API? couldn't find it under zod readme.

Reply:
It has nothing to do with zod, it's just an option to opt-out of the zod transforms within the form. I don't think they should've been there in the first place as zod's `.transform()` can change the data type

Did this as a further explanation: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbwF4F84DMoRHA5EiAE1wFgAoc8gYwgDsBneeqgCwFMQBDOAXjiQB0EAEYArNlRgAKBOTjy4hTjDYBlGFABc-OQoGMowWgHMpASl3yBGzg3TQQUqQDdOAGwCubM7wB8cWjYAdzgAEWU2F3cvMwsyBTgBKDZ0I0ioz28-OABCYHoAOU4CjK8BYzYYABVQSNiAGkRLBRA2enpOCu1cAElaVzdgQkUI3HrmlDNxsknKMhoGeCMwD3g+WXiRlXUtOAAiACYABmOAWiOARnOLvemUAG45hcY4ZPpeOGZ2Ln1OdDYAAqcKD0SLLVZxObAdBwKRvfQeKhUNr0HwbeQAegxcCqLHycHxMHY-CIrzaEDczjYUAJ7w89CMxlewGMLHgtAgQV0zwpbAEbggpnhShgnDimOxCSlUoAegB+XRYuA9eBBYBuNxwGx2Bxa4kGRkE2gwCBwbjhFTcuj0Xn8wVwtoCEWcJ0RHbiuBK6Xe6XyuYoOZKtVEuAQDw0qCcIIANWibVDYBgwGtcHYyUUEHjhNNRkYbE4w3pbDgAANwTAS9RrbaBaZy3EvT64H6yEqABLU4tE4sitQaMlcXNwYRsQ3cA0mOBBPGsAlLd4j+zp7tayM6qAgKsMGv28uu7YaBuSpuyuVAA

Reply:
sorry, I am blind... I saw you added a new config.

Reply:
If we are adding this new config, it probably should apply to all resolvers, which I am happy to look into this and get documented as well.

Reply:
Thanks, @KATT I think the config name is great. 👍

Reply:
I had a look around and there's inconsistency how the resolvers behave in terms of this

Here's a few that return the raw value as default:

- https://github.com/react-hook-form/resolvers/blob/350d255817de8820aa62e6f2036daff211708d1e/vest/src/vest.ts#L50
- https://github.com/react-hook-form/resolvers/blob/350d255817de8820aa62e6f2036daff211708d1e/typanion/src/typanion.ts#L37
- https://github.com/react-hook-form/resolvers/blob/350d255817de8820aa62e6f2036daff211708d1e/nope/src/nope.ts#L45
- https://github.com/react-hook-form/resolvers/blob/350d255817de8820aa62e6f2036daff211708d1e/class-validator/src/class-validator.ts#L61

Reply:
If you want to avoid breaking changes, we can't have a default behavior for this. I think it's better to only add it to zod for now.

Reply:
great find 👍 let's keep them consistent. ❤️

Reply:
1. Do you want this to be done ahead of merging this PR?
2. What behavior do you want as default?

Reply:
The default behavior should return by what returned by the resolver 

```tsx
rawValues: false
```

and users can opt in to return rawValues if they want the values to return by hook form. I am happy to merge this PR to get started just let @jorisre do a review on this as well.